lib/manageiq/appliance_console/auth_utilities.rb

```diff
@@ -0,0 +1,88 @@
+require "uri"
+require "erb"
+
+module ManageIQ
+  module ApplianceConsole
+    module AuthUtilities
+      HTTPD_CONFIG_DIRECTORY = Pathname.new("/etc/httpd/conf.d")
+
+      def restart_httpd
+        httpd_service = LinuxAdmin::Service.new("httpd")
+        if httpd_service.running?
+          say("Restarting httpd ...")
+          httpd_service.restart
+        end
+      end
+
+      def path_is_file?(path)
+        path.present? && !path_is_url?(path)
+      end
+
+      def path_is_url?(path)
+        path =~ /\A#{URI.regexp(["http", "https"])}\z/x
+      end
+
+      # File Management
+
+      def remove_file(path)
+        if path.exist?
+          debug_msg("Removing #{path} ...")
+          File.delete(path)
+        end
+      end
+
+      def copy_template(dir, file, template_parameters = nil)
+        src_path = template_directory.join(relative_from_root(dir), file)
+        dest_path = dir.join(file)
+        dest_path = dest_path.sub_ext('') if src_path.extname == ".erb"
+        debug_msg("Copying template #{src_path} to #{dest_path} ...")
+        if src_path.extname == ".erb"
```

Comment:
I don't understand why there's .erb templating here, but not with other things like saml.

Reply:
SAML auth handshaking (certs/keys) is file driven, OIDC id/secrets are in the conf file so erb to the rescue.